zbus/src/connection/mod.rs

```diff
@@ -1534,6 +1534,81 @@ mod p2p_tests {
         )
     }
 
+    #[cfg(any(
+        all(feature = "vsock", not(feature = "tokio")),
+        feature = "tokio-vsock"
+    ))]
+    #[test]
+    #[timeout(15000)]
+    fn vsock_connect() {
+        crate::utils::block_on(test_vsock_connect()).unwrap();
+    }
+
+    #[cfg(all(feature = "vsock", not(feature = "tokio")))]
+    async fn test_vsock_connect() -> Result<()> {
+        let guid = Guid::generate();
+
+        let listener =
+            vsock::VsockListener::bind_with_cid_port(vsock::VMADDR_CID_LOCAL, u32::MAX).unwrap();
+        let addr = listener.local_addr().unwrap();
+        let addr = format!("vsock:cid={},port={},guid={guid}", addr.cid(), addr.port());
+
+        let server = std::thread::spawn(move || {
+            let server = listener.incoming().next().unwrap().unwrap();
+            crate::utils::block_on(async {
+                Builder::vsock_stream(server)
+                    .server(guid)
+                    .unwrap()
+                    .p2p()
+                    .auth_mechanism(AuthMechanism::Anonymous)
+                    .build()
+                    .await
+            })
+            .unwrap()
+        });
+
+        let _conn = crate::connection::Builder::address(addr.as_str())?
+            .p2p()
+            .build()
+            .await
+            .unwrap();
+        let _server = server.join().unwrap();
+
+        Ok(())
+    }
+
+    #[cfg(feature = "tokio-vsock")]
+    async fn test_vsock_connect() -> Result<()> {
+        let guid = Guid::generate();
+
+        let listener = tokio_vsock::VsockListener::bind(1, u32::MAX).unwrap();
+        let addr = listener.local_addr().unwrap();
+        let addr = format!("vsock:cid={},port={},guid={guid}", addr.cid(), addr.port());
+
+        let server = std::thread::spawn(move || {
+            crate::utils::block_on(async {
```

Comment:
why? spawn a task. might be the problem we're seeing (would be strange though). See [here](https://github.com/dbus2/zbus/blob/d87d4fc1dfa5a20be91bc7befe39aa73606a7db4/zbus/src/lib.rs#L1100).

Although I'd [join the futures](https://github.com/dbus2/zbus/blob/99e48fdd9d975e30331803f8bca10577d1024b93/zbus/src/connection/mod.rs#L1500).

Reply:
yes it works without extra thread, I'll update the MR.